palace.pyx

```diff
@@ -543,8 +577,30 @@ cdef class Source:
     def offset(self, value: int) -> None:
         self.impl.set_offset(value)
 
-    # TODO: sample latency
-    # TODO: sec offset and latency
+    @property
+    def sample_latency(self) -> int:
+        """The source offset in sample frames and its latency in nanoseconds.
+
+        If the AL_SOFT_source_latency extension is unsupported, the latency will
+        be 0.
```

Comment:
This is OK, but personally I'd prefer

    If the AL_SOFT_source_latency extension is unsupported,
    the latency will be 0.